test/check-machines-consoles

```diff
@@ -508,6 +508,77 @@ fullscreen=0
         self.performAction(name, "run")
         b.wait_visible(".vm-console-vnc canvas")
 
+    def testScaleResize(self):
+        b = self.browser
+
+        name = "subVmTest1"
+        self.createVm(name, graphics="vnc")
+
+        self.login_and_go("/machines")
+        self.waitPageInit()
+        self.waitVmRow(name)
+        self.goToVmPage(name)
+
+        # The "width" and "height" HTML attributes of the NoVNC canvas
+        # always reflect the size of the remote framebuffer.  If we
+        # request a remote resize and it is rejected, these attributes
+        # will not change.  Thus, we don't actually have to log into
+        # the guest and do stuff like looking at
+        # /sys/class/drm/<foo>/modes to see how big the actual
+        # framebuffer is.
+
+        # Local scaling is done by NoVNC by setting the width and
+        # height style properties of the canvas element.  The browser
+        # will then scale the content of the canvas to make it as
+        # large as requested by its style.
```

Comment:
That's actually rather clever! I had expected noVNC to do the local scaling, but right, I suppose browsers are actually good at that.